### PR TITLE
Move API sample code to tutorial_code location

### PR DESCRIPTION
**Pull request by alexgiving**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Build configuration
```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2021.4.1:20.04
build_image:Custom Win=openvino-2021.4.1
build_image:Custom Mac=openvino-2021.4.1

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Review by OrestChura — APPROVED** [review]
LGTM, thanks!

**Comment by alexgiving** [comment]
@alalek could you merge it please

File: modules/gapi/doc/20-kernel-api.markdown [hunk 1]
```diff
@@ -47,7 +47,7 @@ an external parameter.
 G-API provides a macro to define a new kernel interface --
 G_TYPED_KERNEL():
 
-@snippet modules/gapi/samples/kernel_api_snippets.cpp filter2d_api
+@snippet samples/cpp/tutorial_code/gapi/kernel_api_snippets/kernel_api_snippets.cpp filter2d_api
```
Comment (dmatveev) [inline modules/gapi/doc/20-kernel-api.markdown:50]: > kernel_api_snippets

I don't think we need an individual directory for every of the snippet files. You can place them all in just `doc_snippets` directory and make it once. The `tutorial_code/gapi` directory now may look overloaded with details.

Comment (alexgiving) [reply modules/gapi/doc/20-kernel-api.markdown:50]: Ok, will move all snippets files to tutorial_code/gapi/doc_snippets

**Review by OrestChura — APPROVED** [review]

**Review by dmatveev — APPROVED** [review]
👍 , thanks!

**Comment by alexgiving** [comment]
@alalek please merge the PR

**Pull request merged**